qa/SidechainTestFramework/scutil.py

```diff
@@ -81,153 +88,284 @@ def sync_sc_mempools(api_connections, wait_for=25):
-        'WALLET_SEED' : ("minerNode" + str(n+1)) if n < 3 else ("node" + str(n+1)),
-        'API_ADDRESS' : "127.0.0.1",
-        'API_PORT' : str(apiPort),
-        'BIND_PORT' : str(bindPort),
-        'OFFLINE_GENERATION' : "false",
-        'GENESIS_DATA': "" if genesisData is None else str(genesisData),
-        'GENESIS_SECRETS': genesis_secrets[n]
-        }
+        'NODE_NUMBER': n,
+        'DIRECTORY': dirname,
+        'WALLET_SEED': "sidechain_seed_{0}".format(n),
+        'API_ADDRESS': "127.0.0.1",
+        'API_PORT': str(apiPort),
+        'BIND_PORT': str(bindPort),
+        'OFFLINE_GENERATION': "false",
+        'GENESIS_SECRETS': genesis_secrets[n] if account_secrets==None else ','.join(['"{0}"'.format(value) for value in genesis_secrets])[1:-1],
+        'SIDECHAIN_ID': jsonNode["scId"],
+        'GENESIS_DATA': jsonNode["scGenesisBlockHex"],
+        'POW_DATA': jsonNode["powData"],
+        'BLOCK_HEIGHT': jsonNode["mcBlockHeight"],
+        'NETWORK': str(jsonNode["mcNetwork"])
+    }
     configsData.append({
-        "name" : "node" + str(n),
-        "url" : "http://" + apiAddress + ":" + str(apiPort)
+        "name": "node" + str(n),
+        "url": "http://" + apiAddress + ":" + str(apiPort)
     })
-    with open(os.path.join(datadir, "node"+str(n)+".conf"), 'w+') as configFile:
+    with open(os.path.join(datadir, "node" + str(n) + ".conf"), 'w+') as configFile:
         configFile.write(config)
     return configsData
 
+
+def sc_generate_genesis_data(self):
+    return generate_genesis_data(self.nodes[0])  # Maybe other parameters in future
+
+
 def initialize_sc_chain(test_dir, genesisData):
     pass
 
-def initialize_sc_chain_clean(test_dir, num_nodes, genesisData):
+
+def initialize_sc_chain_clean(test_dir, num_nodes, account_secrets=None, genesis_info=None):
     """
     Create an empty blockchain and num_nodes wallets.
     Useful if a test case wants complete control over initialization.
     """
     for i in range(num_nodes):
-        initialize_sc_datadir(test_dir, i, genesisData)
+        initialize_sc_datadir(test_dir, i, account_secrets, genesis_info)
+
 
-def start_sc_node(i, dirname, extra_args=None, rpchost=None, timewait=None, binary = None):
+def start_sc_node(i, dirname, extra_args=None, rpchost=None, timewait=None, binary=None):
     """
     Start a SC node and returns API connection to it
     """
-    #Will we have  extra args for SC too ?
-    datadir = os.path.join(dirname, "sc_node"+str(i))
+    # Will we have  extra args for SC too ?
+    datadir = os.path.join(dirname, "sc_node" + str(i))
 
     if binary is None:
-        binary = "../examples/simpleapp/target/Sidechains-SDK-simpleapp-0.1-SNAPSHOT.jar;../examples/simpleapp/target/lib/* com.horizen.examples.SimpleApp"
-#        else if platform.system() == 'Linux':
+        binary = "../examples/simpleapp/target/Sidechains-SDK-simpleapp-0.1-SNAPSHOT.jar:../examples/simpleapp/target/lib/* com.horizen.examples.SimpleApp"
```

Comment:
Maybe, after Linux fix it will brake Windows

Reply:
Ok. make it work for both platforms.
Note: As we discussed before, external libs dependencies should be removed later.

Reply:
Done